tools/scripts/deps-download/download-dep.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import urllib.parse
+import urllib.request
+import uuid
+from pathlib import Path
+
+# Setup logging
+# Create logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+# Setup console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter("%(asctime)s [%(levelname)s] %(message)s")
+logging_console_handler.setFormatter(logging_formatter)
+logger.addHandler(logging_console_handler)
+
+
+def main(argv):
+    args_parser = argparse.ArgumentParser(description="Download dependency.")
+    args_parser.add_argument("source_url", help="Url to the source file.")
+    args_parser.add_argument("source_name", help="Name of the source file.")
+    args_parser.add_argument("dest_dir", help="Destination to output the downloaded dependency.")
+    args_parser.add_argument("--extract", action="store_true", help="Extract the source file.")
+    args_parser.add_argument(
+        "--no-submodule",
+        action="store_false",
+        dest="use_submodule",
+        help="Do not use git submodule update",
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+    target_dest_path = Path(parsed_args.dest_dir).resolve()
+    source_url = parsed_args.source_url
+    source_name = parsed_args.source_name
+    extract_source = parsed_args.extract
+
+    script_path = Path(os.path.realpath(__file__))
+    git_dir = script_path.parent / ".." / ".." / ".." / ".git"
+    if git_dir.exists() and git_dir.is_dir():
+        if parsed_args.use_submodule:
+            cmd = ["git", "submodule", "update", "--init", str(target_dest_path)]
+            try:
+                subprocess.run(cmd, check=True)
+            except subprocess.CalledProcessError:
+                logger.exception(f"Failed to update the submodule {target_dest_path}")
+                return -1
+            return 0
+
+    work_dir = Path("/") / "tmp" / str(uuid.uuid4())
+    work_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
```suggestion
    with tempfile.TemporaryDirectory() as work_dir_name:
        work_dir = Path(work_dir_name)
```

And then indent everything below up to the return statement.

Also update the imports as necessary.